src/main/java/frc/robot/vision/multivisionsources/MultiAprilTagVisionSources.java

```diff
@@ -0,0 +1,91 @@
+package frc.robot.vision.multivisionsources;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.vision.data.AprilTagVisionData;
+import frc.robot.vision.sources.GyroRequiringVisionSource;
+import frc.robot.vision.GyroAngleValues;
+import frc.robot.vision.sources.VisionSource;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class MultiAprilTagVisionSources extends MultiVisionSources<AprilTagVisionData> {
+
+	private final Supplier<Rotation2d> gyroSupplier;
+	private final Supplier<Rotation2d> headingOffsetSupplier;
+	private boolean useGyroForPoseEstimating;
+
+	@SafeVarargs
+	public MultiAprilTagVisionSources(
+		String logPath,
+		Supplier<Rotation2d> gyroSupplier,
+		Supplier<Rotation2d> headingOffsetSupplier,
+		VisionSource<AprilTagVisionData>... visionSources
+	) {
+		this(logPath, gyroSupplier, headingOffsetSupplier, List.of(visionSources));
+	}
+
+	public MultiAprilTagVisionSources(
+		String logPath,
+		Supplier<Rotation2d> gyroSupplier,
+		Supplier<Rotation2d> headingOffsetSupplier,
+		List<VisionSource<AprilTagVisionData>> visionSources
+	) {
+		super(logPath, visionSources);
+		this.gyroSupplier = gyroSupplier;
+		this.headingOffsetSupplier = headingOffsetSupplier;
+		this.useGyroForPoseEstimating = true;
```

Comment:
```suggestion
		this.useGyroForPoseEstimating = true;
```

gyro for eveyrhing? true just for the limelight. usebotpose1 is a better name. or usegyroforlimelightestimating